### PR TITLE
feat: Create stream and batch feature view abstractions

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
in prep for upcoming streaming work

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/batch_feature_view.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from datetime import timedelta
+from typing import Dict, List, Optional, Union
+
+from google.protobuf.duration_pb2 import Duration
+
+from feast import Entity, Feature, Field
```
Comment (felixwang9817) [inline sdk/python/feast/batch_feature_view.py:6]: nit: elsewhere we tend to do `from feast.entity import Entity` etc.

we should try to stay consistent

File: sdk/python/feast/batch_feature_view.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "SnowflakeSource",
+    "SparkSource",
+    "TrinoSource",
+}
+
+
+class BatchFeatureView(FeatureView):
+    def __init__(
+        self,
+        *,
```
Comment (felixwang9817) [inline sdk/python/feast/batch_feature_view.py:24]: nice

File: sdk/python/feast/batch_feature_view.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        name: Optional[str] = None,
+        entities: Optional[Union[List[Entity], List[str]]] = None,
+        ttl: Optional[Union[Duration, timedelta]] = None,
+        features: Optional[List[Feature]] = None,
```
Comment (felixwang9817) [inline sdk/python/feast/batch_feature_view.py:28]: is BFV meant to be backwards compatible with FV? if not, we should just eliminate this now and switch to `schema`

and make all the changes to other parameters, e.g. `ttl` should not support a `Duration` object

File: sdk/python/feast/stream_feature_view.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+from datetime import timedelta
+from typing import Dict, List, Optional, Union
+
+from google.protobuf.duration_pb2 import Duration
+
+from feast import Entity, Feature, Field
```
Comment (felixwang9817) [inline sdk/python/feast/stream_feature_view.py:6]: all the comments from BFV apply here

File: sdk/python/tests/unit/test_feature_views.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
+from datetime import timedelta
+
+import pytest
+
+from feast import KafkaSource
```
Comment (felixwang9817) [inline sdk/python/tests/unit/test_feature_views.py:5]: same comment

File: sdk/python/tests/unit/test_feature_views.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+from feast.batch_feature_view import BatchFeatureView
+from feast.data_format import AvroFormat
+from feast.infra.offline_stores.file_source import FileSource
+from feast.stream_feature_view import StreamFeatureView
+
+
+def test_create_batch_feature_view():
+    batch_source = FileSource("some path")
```
Comment (felixwang9817) [inline sdk/python/tests/unit/test_feature_views.py:13]: nit: use kwarg?

**Review by felixwang9817 — COMMENTED** [review]
some nits

File: sdk/python/feast/batch_feature_view.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
+from datetime import timedelta
+from typing import Dict, List, Optional, Union
+
+from google.protobuf.duration_pb2 import Duration
```
Comment (woop) [inline sdk/python/feast/batch_feature_view.py:4]: Do we still need to support Duration? Isn't it a good time to deprecate it?

File: sdk/python/feast/batch_feature_view.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/batch_feature_view.py:6]: good catch

File: sdk/python/feast/batch_feature_view.py (hunk 3 above)
Comment (achals) [reply sdk/python/feast/batch_feature_view.py:28]: good catch. The new classes don't need to support the old API.

File: sdk/python/tests/unit/test_feature_views.py (hunk 6 above)
Comment (achals) [reply sdk/python/tests/unit/test_feature_views.py:13]: lol yea

File: sdk/python/feast/batch_feature_view.py (hunk 7 above)
Comment (achals) [reply sdk/python/feast/batch_feature_view.py:4]: yes it is. Removed

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**